src/git.rs

```diff
@@ -79,40 +115,197 @@ impl Git for LiveGit {
             None => Err(Box::new(git2::Error::from_str("No branch name found")))
         }
     }
-}
 
-#[cfg(test)]
-mod test {
-    use super::*;
-    
-    pub struct MockGit;
-    impl Git for MockGit {
-        fn new() -> Result<Self, Box<dyn Error>> {
-            Ok(MockGit)
-        }
-        fn read_file(&self, _path: &str) -> Result<String, Box<dyn Error>> {
-            Ok(String::from(""))
+    fn log(&self) -> Result<(), Box<dyn Error>> {
+        let mut revwalk = self.repo.revwalk()?;
+        revwalk.push_head()?;
+        for commit_id in revwalk {
+            let commit = self.repo.find_commit(commit_id?)?;
+            Self::debug_commit(&commit);
         }
-        fn current_branch(&self) -> Result<String, Box<dyn Error>> {
-            Ok(String::from("master"))
+
+        Ok(())
+    }
+
+    fn find_commit(&self, commit_id: Oid) -> Result<Commit<'_>, Box<dyn Error>> {
+        Ok(self.repo.find_commit(commit_id)?)
+    }
+
+    fn find_commit_fingerprints(&self, team_fingerprint_file: &str, commits: &Vec<Commit<'_>>) -> Result<HashSet<String>, Box<dyn Error>> {
+
+        let team_fingerprints = Fingerprint::read_fingerprints::<LiveGit>(self, team_fingerprint_file)?;
+
+        let mut commit_fingerprints = HashSet::new();
+
+        for commit in commits.iter() {
```

Comment:
You could turn this into two successive calls to map to avoid the mutable variable above.